ios/Classes/utils/applePayConfiguration/ApplePayConfigurationParser.swift

```diff
@@ -0,0 +1,286 @@
+        let paymentRequest = try buildPaymentRequest(amount: amount, countryCode: countryCode)
+        return try ApplePayComponent.Configuration(paymentRequest: paymentRequest, allowOnboarding: allowOnboarding ?? false)
+    }
+    
+    private func buildPaymentRequest(amount: AmountDTO, countryCode: String) throws -> PKPaymentRequest {
+        let paymentRequest = PKPaymentRequest()
+        paymentRequest.merchantIdentifier = merchantId
+        paymentRequest.paymentSummaryItems = try mapToPaymentSummaryItems(summaryItems: summaryItems, amount: amount)
+        paymentRequest.countryCode = countryCode
+        paymentRequest.currencyCode = amount.currency
+        paymentRequest.billingContact = billingContact?.toApplePayContact()
+        paymentRequest.shippingContact = shippingContact?.toApplePayContact()
+        paymentRequest.merchantCapabilities = merchantCapability.toMerchantCapability()
+        
+        if let requiredShippingContactFields {
+            paymentRequest.requiredShippingContactFields = mapToContactFields(contactFields: requiredShippingContactFields)
+        }
+        
+        if let requiredBillingContactFields = requiredShippingContactFields {
+            paymentRequest.requiredBillingContactFields = mapToContactFields(contactFields: requiredBillingContactFields)
+        }
+        
+        if let shippingType = applePayShippingType {
+            paymentRequest.shippingType = shippingType.toPKShippingType()
+        }
+        
+        if let supportedCountries {
+            paymentRequest.supportedCountries = .init(supportedCountries.compactMap { $0 })
+        }
+        
+        if let shippingMethods {
+            paymentRequest.shippingMethods = try shippingMethods.compactMap { try $0?.toPKShippingMethod() }
+        }
+        
+        if #available(iOS 15.0, *) {
+            if let allowShippingContactEditing {
+                // We have to use enabled until we forcing the newest Xcode version. Otherwise the build fails.
+                paymentRequest.shippingContactEditingMode = allowShippingContactEditing == true ? PKShippingContactEditingMode.enabled : PKShippingContactEditingMode.storePickup
+            }
+        }
+        
+        if let supportedNetworks {
+            let supportedNetworksNonNil: [String] = supportedNetworks.compactMap { $0 }
+            paymentRequest.supportedNetworks = mapToSupportedNetworks(supportedNetworks: supportedNetworksNonNil)
+        }
+        
+        if let applicationData {
+            paymentRequest.applicationData = Data(applicationData.utf8)
+        }
+
+        return paymentRequest
+    }
+    
+    private func addShippingMethodToSummaryItems(paymentRequest: PKPaymentRequest) {
+        guard let shippingMethod = paymentRequest.shippingMethods?.first else {
+            return
+        }
+        
+        if let last = paymentRequest.paymentSummaryItems.last {
```

Comment:
Maybe you want to write a comment on what this is actually doing? It's not immediately clear